shotgun_api3/shotgun.py

```diff
@@ -1867,14 +1888,15 @@ def _response_errors(self, sg_response):
 
         :raises ShotgunError: If the server response contains an exception.
         """
- 
+
         ERR_AUTH = 102 # error code for authentication related problems
+        ERR_2FA  = 106 # error code when 2FA authentication is required but no 23FA token provided.
 
         if isinstance(sg_response, dict) and sg_response.get("exception"):
-            
             if sg_response.get("error_code") == ERR_AUTH:
                 raise AuthenticationFault(sg_response.get("message", "Unknown Authentication Error"))
-
+            elif sg_response.get("error_code") == ERR_2FA:
+                raise MultiFactorAuthenticationFault(sg_response.get("message", "Unknown 2FA Authentication Error"))
```

Comment:
The Authentication error is unknown because it is a generic catch all error for any of various authentication problems.  Is there a better error message possible for this error code?  Also, is there a reason to use 2FA here rather than Multifactor?  The wording in the error code variable/fault and the error message seems inconsistent.